src/main/java/internet/market/dao/jdbc/OrderDaoJdbcImpl.java

```diff
@@ -0,0 +1,193 @@
+package internet.market.dao.jdbc;
+
+import internet.market.dao.OrderDao;
+import internet.market.exceptions.DataProcessingException;
+import internet.market.lib.Dao;
+import internet.market.model.Order;
+import internet.market.model.Product;
+import internet.market.util.ConnectionUtil;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Dao
+public class OrderDaoJdbcImpl implements OrderDao {
+
+    @Override
+    public List<Order> getUserOrders(Long userId) {
+        String query = "SELECT * FROM orders WHERE user_id = ? AND deleted = false";
+        List<Order> orderList = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement preparedStatement = connection.prepareStatement(query);
+            preparedStatement.setLong(1, userId);
+            ResultSet resultSet = preparedStatement.executeQuery();
+            while (resultSet.next()) {
+                orderList.add(getOrderInformationFromDataBase(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can`t get orders of user with id = " + userId, e);
+        }
+        for (Order order : orderList) {
+            order.setProducts(getProductsOfOrder(order.getId()));
+        }
+        return orderList;
+    }
+
+    @Override
+    public Order create(Order item) {
+        String query = "INSERT INTO orders(user_id) VALUES (?)";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query,
+                    Statement.RETURN_GENERATED_KEYS);
+            statement.setLong(1, item.getUserId());
+            statement.executeUpdate();
+            ResultSet resultSet = statement.getGeneratedKeys();
+            if (resultSet.next()) {
+                item.setId(resultSet.getLong(1));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can`t add order with id =  " + item.getId(), e);
+        }
+        addProductsToOrder(item.getProducts(), item.getId());
+        return item;
+    }
+
+    @Override
+    public Optional<Order> get(Long id) {
+        String query = "SELECT * FROM orders WHERE order_id = ? AND deleted = false";
+        Order order = null;
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                order = getOrderInformationFromDataBase(resultSet);
+                order.setProducts(getProductsOfOrder(id));
+                return Optional.of(order);
```

Comment:
do we need to return here if we decide to remove the previous line?

Reply:
done